### PR TITLE
[testslide-snippets] Add vscode snippets extension

### PR DESCRIPTION
**Pull request by david-caro**
DO NOT MERGE: still under development

**What:**

This adds the code for a VSCode extension adding some helpful snippets
to develop with TestSlide.

**Why:**

This will help develop tests faster, improving considerably the developer experience (in most cases).

**Checklist**:

- [x] Added tests, if you've added code that should be tested
- [ ] Updated the documentation, if you've changed APIs
- [x] Ensured the test suite passes
- [x] Made sure your code lints
- [x] Completed the Contributor License Agreement ("CLA")

### CODE REVIEW

**Comment by cricalix** [comment]
Some thoughts:

1. All of the examples are for non-async methods. I was going to suggest tsa, but tsa is already taken as a prefix.
2. It may make sense to use the TM_SELECTED_TEXT fragment in things like the snippet for creating an example. Someone might be porting an existing block of code, and if they've selected some text - Ctrl-Shift-P, Insert Snippet<enter>, arrow key to relevant snippet - will let them just wrap the selection. It takes a default, so you can do ${TM_SELECTED_TEXT:default_if_no_selection}.
3. From the documented nomenclature, tsb_full should be tsab_full, because it has before and after?

File: util/testslide-snippets/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Change Log
+## 1.0.0
```
Comment (fornellas) [inline util/testslide-snippets/CHANGELOG.md:2]: @david-caro  do you think it makes sense to sync this with TestSlide's version? If yes, we should also update `RELEASE.md` to cover this.

File: util/testslide-snippets/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+The snippets follow some common guidelines:
+
+- All snippets are prefixed with `ts` (for TestSlide).
+- Snippets starting with `tsb_` will generate basic test cases and add a before clause.
+- Snippets starting with `tsa_` will generate basic test cases and add an after clause.
+- Snippets starting with `tsab_` will generate basic test cases with both before and after clause.
+- Snippets starting with `tsab_` will generate basic async test cases and add a before clause.
+- Snippets starting with `tsaa_` will generate basic async test cases and add an after clause.
+- Snippets starting with `tsaab_` will generate basic async test cases with both before and after clause.
```
Comment (fornellas) [inline util/testslide-snippets/README.md:15]: I think this is bound to get out of sync with snippets. How about just keep the "everything is ts*"?

File: util/testslide-snippets/CHANGELOG.md (hunk 1 above)
Comment (david-caro) [reply util/testslide-snippets/CHANGELOG.md]: Done

**Comment by fornellas** [comment]
This is good to merge, right @david-caro ?

**Comment by david-caro** [comment]
Now it is, added the note on how to publish the extension :)

**Comment by david-caro** [comment]
The coverage seems the same, the report is unable to tell me what changed exactly

**Pull request merged**